pytorch_lightning/strategies/horovod.py

```diff
@@ -184,8 +196,13 @@ def post_backward(self, closure_loss: torch.Tensor) -> None:
 
     def _wrap_optimizers(self, optimizers: List[Optimizer]) -> List["hvd.DistributedOptimizer"]:
         """Wraps optimizers to perform gradient aggregation via allreduce."""
+        accumulate_grad_batches = self.trainer.accumulate_grad_batches
```

Comment:
can you add a test?

Reply:
Sure, makes sense. Getting a weird error with `horovod.torch` not available from the helper function `_module_available` in utils. Working on it. :)

Reply:
Tried adding a test for both CPU and GPU devices, with `accumulate_grad_batches` as an argument in the constructor to trainer. Also minor fix, as it should be `self.lightning_module.trainer` instead of `self.trainer`. I had to explicitly set `_HOROVOD_AVAILABLE` as `True` to run tests locally, for some reason `_module_available` returns `False` for `horovod.torch`.